clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -2480,6 +2483,11 @@ public void testDropCommitOnBatchExpiry() throws InterruptedException {
         TransactionalRequestResult abortResult = transactionManager.beginAbort();
 
         prepareEndTxnResponse(Errors.NONE, TransactionResult.ABORT, pid, epoch);
+
+        sender.runOnce();  // send the abort.
```

Comment:
With the call to `runUntil`, we can get rid of these individual calls to `runOnce`. For example:
```java
prepareEndTxnResponse(Errors.NONE, TransactionResult.ABORT, pid, epoch);
prepareInitPidResponse(Errors.NONE, false, pid, (short) (epoch + 1));
runUntil(abortResult::isCompleted);
```
Similarly for the other test cases.

Reply:
Cleaned up a bunch of the test cases to use `runUntil`